packages/ui/src/components/TradingForm.tsx

```diff
@@ -0,0 +1,106 @@
+import { Button, FormikTextField } from '@uzh-bf/design-system'
+import { Form, Formik } from 'formik'
+import * as yup from 'yup'
+
+// TODO(JJ):
+// - UserNotification as child?
+// - Name is currently Trading -> input spotPrice, how to name?
+
+function optionalValueToCHFString(value: number, digits = 2) {
+  return value?.toLocaleString('de-CH', {
+    style: 'currency',
+    currency: 'CHF',
+    maximumFractionDigits: digits,
+  })
+}
+
+interface Props {
+  price: number
+  nameButtonBuy: string
+  nameButtonSell: string
+  onSubmit: (values: any, helpers: any) => Promise<void>
+  min?: number
+  disableButtonBuy?: boolean
+  disableButtonSell?: boolean
+}
+
+function TradingForm({
+  price,
+  nameButtonBuy,
+  nameButtonSell,
+  onSubmit,
+  min = 0,
+  disableButtonBuy = false,
+  disableButtonSell = false,
+}: Props) {
+  return (
+    <div className="flex w-max gap-4 rounded border p-8">
+      <Formik
+        initialValues={{
+          modifier: 1,
+          volume: min,
```

Comment:
why is min the initial volume? should it not be 0?

Reply:
I agree that this value shouldn't be lower than 0. However, I wanted to keep it generic: e.g. when it is a requirement that you need to trade for a minimun volume (min >=0). I will check if I can do a validation, s.t. the input min >=0.

Reply:
Changed it to 0. We can adapt that if requried.